kafka-connect/src/main/java/io/tabular/iceberg/connect/api/Committer.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.tabular.iceberg.connect.api;
+
+public interface Committer {
+  void commit(CommittableSupplier committableSupplier);
```

Comment:
A committer is responsible for taking a committable and committing it to iceberg.

Reply:
*nit, see naming.  `CommitterImpl` doesn't take a `CommitableSupplier` and commit to iceberg.  It produces Commit events to Kafka / commits kafka offsets, but it is the `Coordinator` (which is not a subclass of Committer) that commits to iceberg.  I've already mentioned the naming is a bit confusing.

Reply:
> CommitterImpl doesn't take a CommitableSupplier and commit to iceberg.

Actually, that's exactly what it does :D Bear with me for a second ...

> It produces Commit events to Kafka / commits kafka offsets, but it is the Coordinator (which is not a subclass of Committer) that commits to iceberg. 

The fact that `CommitterImpl` produces events to Kafka that the `Coordinator` then picks up and commits to iceberg is purely an implementation detail of HOW this `Committer` interface implementation works. The Coordinator is not a subclass of `Committer` and does not need to be. It is a private implementation detail of `CommiterImpl`. 

You can have other `Committer` interface implementations in the future that work completely differently (i.e. have no `Coordinator` class). 

At the end of the day, a `Committer` interface implementation must `take a Commitable and commit it to iceberg`, whether it does that via some kafka-channel-coordinator-class-mechanism or whatever is implementation details. 

---

All that being said, I can definitely see how things might be confusing but I would argue the the `CommitterImpl` is what is confusing, and not the naming of the `Committer` interface.  Happy to go over this offline, I think this discussion might be a bit too nuanced for words-over-github.

Also open to suggestions on naming :)

Reply:
I can live with this.  I have no better suggestion on naming.